scipy/sparse/sparsetools/coo.h

```diff
@@ -112,6 +112,53 @@ void coo_todense(const I n_row,
 }
 
 
+/*
+ * Input Arguments:
+ *   I  shape[num_dims]  - shape of A
+ *   npy_int64  nnz      - number of nonzeros in A
+ *   npy_int64  num_dims - number of dimensions of A
+ *   I  Aijk[nnz(A)]     - coords for nonzeros in A
+ *   T  Ax[nnz(A)]       - nonzeros in A
+ *   T  Bx[]             - dense array
+ *
+ */
+template <class I, class T>
+void coo_todense_nd(const I shape[],
```

Comment:
Instead of passing in a `shape[]` array, how about passing in the cumulative product as a `strides[]` array instead? That would avoid having to re-calculate the stride every iteration of the main loop.

Reply:
Just to confirm, we would need to pass both C and Fortran strides for `coo_todense_nd`, right?
Addressed at d56b75ef1728004e53657da892d73ad6bf555eee.

Reply:
We don't actually need both C and F strides. If we look at the value we input for `fortran` we can figure out which strides to input.

Reply:
Thanks, modified at f9ba32125b0a286f2e50a405263d40eded6465a0.